plugins/serialization/src/main/java/org/qbicc/plugin/serialization/BuildtimeHeap.java

```diff
@@ -84,18 +102,25 @@ public GlobalVariableElement getAndRegisterGlobalClassArray(ExecutableElement or
         return classArrayGlobal;
     }
 
-    public synchronized Data serializeStringLiteral(String value) {
-        // String literals are interned via equals, not ==
-        if (stringLiterals.containsKey(value)) {
-            return stringLiterals.get(value);
+    public synchronized Data serializeVmObject(VmObject value) {
+        if (vmObjects.containsKey(value)) {
+            return vmObjects.get(value);
         }
-        LoadedTypeDefinition jls = ctxt.getBootstrapClassContext().findDefinedType("java/lang/String").load();
-        Data sl = serializeObject(jls, value);
-        stringLiterals.put(value, sl);
+        PhysicalObjectType ot = value.getObjectType();
+        Data sl;
+        if (ot instanceof ClassObjectType) {
+            // TODO: Right here we need to detect if value is a java.lang.Class instance and handle it specially by calling serializeClassObject!
```

Comment:
Are we reifying java.lang.Class instances into VmObjects in the interpreter?  Or do we do something else if the build time heap tries to instantiate a Class instance?

Reply:
Cool.  I'll make that adjustment this morning.

Reply:
Added the logic for VmClass.  Ready for review